vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java

```diff
@@ -132,6 +133,7 @@ public void handle(Throwable event) {
                 log.debugf(event, "IO Exception ");
                 //TODO: do we need this?
                 eof = true;
+                responseError = event;
```

Comment:
idle timeout causes the response exception handler to be called (at least to be called first) and when `terminate*` is called and does `notify` on the response, `readBlocking` is triggered.
Without checking for `responseError`, this would lead to partial request passed to servlert/rest endpoint